bin/alias.ml

```diff
@@ -94,15 +94,37 @@ let dep_on_alias_rec_multi_contexts ~dir:src_dir ~name ~contexts =
          in
          Dune_rules.Alias_rec.dep_on_alias_rec name dir))
   in
-  let is_nonempty =
-    List.exists alias_statuses ~f:(fun (x : Action_builder.Alias_status.t) ->
+  match
+    Alias.is_standard name
+    || List.exists alias_statuses ~f:(fun (x : Action_builder.Alias_status.t) ->
       match x with
       | Defined -> true
       | Not_defined -> false)
-  in
-  if (not is_nonempty) && not (Alias.is_standard name)
-  then
+  with
+  | true -> Action_builder.return ()
+  | false ->
+    let* load_dir =
+      Action_builder.all
+      @@ List.map contexts ~f:(fun ctx ->
+        let dir =
+          Source_tree.Dir.path dir
+          |> Path.Build.append_source (Context_name.build_dir ctx)
+          |> Path.build
+        in
+        Action_builder.of_memo @@ Load_rules.load_dir ~dir)
+    in
+    let hints =
+      let candidates =
+        List.fold_left load_dir ~init:Alias.Name.Map.empty ~f:(fun acc x ->
```

Comment:
Actually, I think you can just accumulate a set here. As far as I can tell, you don't need the values of this map so you can discard them right away. We have `Alias.Name.Set.of_keys` for this.